services/ux-backend/handlers/removecephclient/handler.go

```diff
@@ -0,0 +1,125 @@
+	klog.Info("Only POST method should be used to send data to this endpoint /delete-ceph-client")
+	w.WriteHeader(http.StatusMethodNotAllowed)
+	w.Header().Set("Content-Type", handlers.ContentTypeTextPlain)
+	w.Header().Set("Allow", "POST")
+
+	if _, err := w.Write([]byte(fmt.Sprintf("Unsupported method : %s", r.Method))); err != nil {
+		klog.Errorf("failed to write data to response writer: %v", err)
+	}
+}
+
+func removeCephZombieResources(ctx context.Context, cl client.Client, consumerName string) (string, error) {
+
+	storageconsumer := &ocsv1alpha1.StorageConsumer{}
+	err := cl.Get(context.Background(), types.NamespacedName{Name: consumerName,
+		Namespace: handlers.GetPodNamespace()}, storageconsumer)
+	if err != nil {
+		return "", fmt.Errorf("failed to get storage consumer with name: %s:%v", storageconsumer.Name, err)
+	}
+
+	storageclassrequests := &ocsv1alpha1.StorageClassRequestList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(handlers.GetPodNamespace()),
+		client.MatchingLabels{"ocs.openshift.io/storageconsumer-uuid": string(storageconsumer.UID)},
+	}
+	err = cl.List(ctx, storageclassrequests, listOpts...)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to get list of storageclassrequests, %v", err)
+	}
+
+	if len(storageclassrequests.Items) == 0 {
+		return "", fmt.Errorf("no storageclassrequests present")
+	}
+	for _, scr := range storageclassrequests.Items {
+		for _, cephResourceSpec := range scr.Status.CephResources {
+			if cephResourceSpec.Kind == "CephBlockPoolRadosNamespace" {
+				radosNamespace := &cephv1.CephBlockPoolRadosNamespace{}
+				err = cl.Get(context.Background(), types.NamespacedName{Name: cephResourceSpec.Name,
+					Namespace: scr.Namespace}, radosNamespace)
+				if err != nil {
+					return "", fmt.Errorf("failed to get CephBlockPoolRadosNamespace, %v", err)
+				}
+				if radosNamespace.Annotations == nil {
+					radosNamespace.Annotations = make(map[string]string)
+				}
+				radosNamespace.Annotations["ceph.rook.io/force-deletion"] = "true"
+				err := cl.Update(ctx, radosNamespace)
+				if err != nil {
+					return "", fmt.Errorf("failed to annotate CephBlockPoolRadosNamespace, %v", err)
+				}
+			}
+			if cephResourceSpec.Kind == "CephFilesystemSubVolumeGroup" {
+				subVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{}
+				err = cl.Get(context.Background(), types.NamespacedName{Name: cephResourceSpec.Name,
+					Namespace: scr.Namespace}, subVolumeGroup)
+				if err != nil {
+					return "", fmt.Errorf("failed to get CephFilesystemSubVolumeGroup, %v", err)
+				}
+				if subVolumeGroup.Annotations == nil {
+					subVolumeGroup.Annotations = make(map[string]string)
+				}
+				subVolumeGroup.Annotations["ceph.rook.io/force-deletion"] = "true"
+
+				err := cl.Update(ctx, subVolumeGroup)
+				if err != nil {
+					return "", fmt.Errorf("failed to annotate CephFilesystemSubVolumeGroup, %v", err)
+				}
+			}
+		}
+	}
+	err = cl.Delete(ctx, storageconsumer)
```

Comment:
Since deleting a storage consumer triggers a delete event on storage class request, so we don't need to SCR mannually.